02_bubble/willshen8/bubble.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	sliceCopy := append(s[:0:0], s...)
+	sliceLength := len(s)
+	sorted := false
+
+	for !sorted {
+		swapOccurred := false
+		for i := 0; i < sliceLength-1; i++ {
+			if sliceCopy[i+1] < sliceCopy[i] {
+				swap(sliceCopy, i, i+1)
+				swapOccurred = true
+			}
+		}
+		if !swapOccurred {
+			sorted = true
```

Comment:
Had a review comment on my code suggest moving the `return` here. It's kinda neat and removes the need for the `sorted` variable.

Reply:
I couldn't get it to work without making my code confusing. I will definitely take on this advise in the future.